GRADING.md

```diff
@@ -0,0 +1,101 @@
+# Assessment Grading Guide for API Development with Java 17 & SpringBoot
+
+Welcome to the grading guide for the API Development Assessment. This guide has been meticulously designed to ensure a clear, consistent, and comprehensive evaluation of students' projects. As an assessor, you'll find a rubric below that breaks down the evaluation criteria into four distinct sections: 
+- Completion
+- Functionality
+- Code Quality & Structure
+- Language Profiency 
+
+Please go through each section methodically, ensuring that you understand the criteria and can confidently attribute a score from 0 to 5 for each student's work based on their performance in the listed areas.
+
+## Table of Contents
+
+1. [Completion](#completion)
+1. [Functionality](#functionality)
+1. [Code Quality & Structure](#code-quality--structure)
+1. [Language Proficiency](#language-proficiency)
+
+## Completion
+
+This section gauges the progress of the student in terms of repository management and project submission. It's a reflection of commitment and the timely delivery of the project.
+
+| **Score** | **Caption**                                          | **Details**                                        |
+|-----------|------------------------------------------------------|----------------------------------------------------|
+| 0         | No repository created                               | Student has not initialised any repository for the project.                                               |
+| 1         | Repository created, but no code commits             | Repository exists but lacks any code. This indicates no progress on the project.                           |
+| 2         | Completed late without agreed extension             | Project was submitted after the deadline without any prior agreement for extension.                        |
+| 3         | Completed within agreed extension                   | Project was submitted late, but within an agreed-upon extension timeframe.                                 |
+| 4         | Completed within deadline                           | Project was submitted on time, indicating good time management.                                           |
+| 5         | Completed within deadline, with full commit history | Project was submitted on time and showcases a consistent work pattern with a well-maintained commit history.|
+
+### Tips
+
+- **Repository Check**: Verify the existence of the repository and its contents to confirm the student's engagement with the project.
+- **Commit Analysis**: Examine the commit history for patterns of work and consistency, reflecting the student's workflow and time management.
+- **Deadline Adherence**: Assess whether the project was completed on time or if an extension was justified and used effectively.
```

Comment:
Should this be a task for the assessor? Or would they just refer to the table and see whether it indicates that an extension was agreed? 

Would suggest this says: 
- **Deadline Adherence**: Verify whether the project was completed on time or within an agreed deadline